### ISSUE TITLE
Support metrics from other GCP projects

### ISSUE THREAD

**Issue by djfinnoy** [GoogleCloudPlatform/prometheus-engine#154]
It appears that metrics from other GCP projects are filtered away by the managed prometheus instance.

We would like to use prometheus and alertmanager to monitor our Cloud Functions. 
Prometheus is running in our GKE cluster in `current-project`, while some (but not not all) of our Cloud Functions belong in a different GCP project (`other-project`).

I've installed the [prometheus stackdriver exporter](https://github.com/prometheus-community/stackdriver_exporter), and got it working with our managed Prometheus instance.

When I curl the `/metrics` endpoint of the exporter, I find lots of results for both projects.
However, in the GCP Console, I can only find the subset of these metrics where `project_id` matches `current-project`.
The metrics where `project_id` matches `other-project` do not make it into Prometheus, even though they are definitely present under the `/metrics` endpoint

I tried adding the following `collections` to the `OperatorConfig`, but it didn't help:
```
apiVersion: monitoring.googleapis.com/v1alpha1
kind: OperatorConfig
metadata:
  name: config
  namespace: gmp-public
collection:
  filter:
    matchOneOf:
      - '{project_id="current-project"}'
      - '{project_id="other-project"}'
```

Concidentally, I see the following error message in the `collector` logs when attempting to get metrics from `other-project`:
```
collector-lp6gq prometheus ts=2022-03-08T13:03:21.525Z caller=log.go:124 component=gcm_exporter level=error msg="send batch" size=4 err="rpc error: code = PermissionDenied desc = Permission monitoring.timeSeries.create denied (or the resource may not exist)."
```

The service account used by Prometheus only has `roles/monitoring.metricsWriter` for `current-project`. Is it possible that `project_id` is being used under the hood, and that Prometheus is trying to write the metrics to stackdriver in `other-project` instead?

**Comment by lyanco** [comment]
Hey there,

First thing I want to note is that we are actively working on allowing users to query Cloud Function and other free GCP system metrics via PromQL. At that point there will be no need to use the stackdriver exporter to read and rewrite the metrics.

You likely need to do two things:

1. Modify your service account to have monitoring.metricsWriter permissions for both projects, as yes the service will try to write back to GCP using any project_id that it finds on a metric
2. Put both of your projects into one [metrics scope](https://cloud.google.com/monitoring/settings?hl=en) and then query the scoping project. This will fan out your queries to both projects and all will work as expected.

**Comment by djfinnoy** [comment]
Excellent. I think we'll just wait for PromQL support for system metrics. Thank you for the help.

**Comment by areyia** [comment]
Having this same issue where we want one of our projects to be the master project and gather metrics from all other projects. We have managed collection running in all projects and would like all metrics to be available in the master project. I understand that at this point, all I need to do is add all the projects to the metrics scope of the master project.

The [documentation](https://cloud.google.com/monitoring/settings?hl=en) tells us how to do it via the UI and API, and I have verified that they both work. However, it would be preferable if we could do this via configuration - is this supported?
